### PR TITLE
Feat/authorization with jwt

### PR DESCRIPTION
**Pull request by ese-varo**
Implement Authentication/Authorization logic using JSON Web Tokens (JWT)

### CODE REVIEW

File: src/middleware/verifySignUp.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    });
+  });
+};
+
+  next();
+};
+
+const verifySignUp = {
+  checkDuplicateUsernameOrEmail: checkDuplicateUsernameOrEmail
+};
```
Comment (LoginALee) [inline src/middleware/verifySignUp.js:41]: I think that if the key is equal to the value you can do
```
const verifySignUp = {
   checkDuplicateUsernameOrEmail
};
```

**Review by LoginALee — APPROVED** [review]
Good job Alvaro :rocket:

File: src/middleware/verifySignUp.js (hunk 1 above)
Comment (ese-varo) [reply src/middleware/verifySignUp.js:41]: you're right! I'll change it, thanks!

**Review by GregVaz — APPROVED** [review]
Great work :rocket:

**Pull request merged**